tests/framework/Optimizers/GradientDescent/constrain_function.xml

```diff
@@ -0,0 +1,114 @@
+<?xml version="1.0" ?>
+<Simulation verbosity="debug">
+  <TestInfo>
+    <name>framework/Optimizers.ConstrainFunction</name>
+    <author>talbpaul</author>
+    <created>2020-02-05</created>
+    <classesTested>Optimizer</classesTested>
+    <description>
+      This test uses a multidimensional linear function such that the trajectory must pass through
+      a functional constraint to reach the optimal point.
```

Comment:
same as above (I think we should tag this as a requirement)

Reply:
Added R-RM-2 for constraining the space of variables used to find min/max goal functions.

Reply:
wonderful